cirq-rigetti/cirq_rigetti/circuit_sweep_executors.py

```diff
@@ -57,21 +57,17 @@ def _execute_and_read_result(
     Raises:
         ValueError: measurement_id_map references an undefined pyQuil readout region.
     """
-    if memory_map is None:
-        memory_map = {}
-
-    for region_name, values in memory_map.items():
-        if isinstance(region_name, str):
-            executable.write_memory(region_name=region_name, value=values)
-        else:
+    memory_map = memory_map or {}
+    for region_name in memory_map.keys():
+        if not isinstance(region_name, str):
             raise ValueError(f'Symbols not valid for region name {region_name}')
-    qam_execution_result = quantum_computer.qam.run(executable)
+    qam_execution_result = quantum_computer.qam.run(executable, memory_map)
```

Comment:
We need to check for atomic values and wrap them in a list to prevent a breaking change here.

In v3, `Program.write_memory` accepted atomic values for memory regions of length 1. For example, `Program.write_memory("ro", 1)` was valid if `ro` was defined as `DECLARE ro BIT[1]`. In v4, the new `MemoryMap` only accepts sequences of values for a memory region, even if that memory region happens to be length 1.

Reply:
fixed this in the lines above 👍